### PR TITLE
microsoft-edge init at 98.0.1108.56

### PR DESCRIPTION
**Pull request by kuwii**
###### Motivation for this change

Add Microsoft Edge browser package to NixOS based on [Azure Zanculmarktum/overlays](https://gitlab.com/zanc/overlays/-/tree/master/edge) authored by @zanculmarktum .

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/browsers/microsoft-edge/browser.nix [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+      --replace "''${XDG_CONFIG_DIRS:-/etc/xdg}" "''${XDG_CONFIG_DIRS:-/run/current-system/sw/etc/xdg}"
+  '';
+
+  meta = with lib; {
+    homepage = "https://www.microsoft.com/en-us/edge";
+    description = "The web browser from Microsoft";
+    license = licenses.unfree;
+    platforms = [ "x86_64-linux" ];
+    maintainers = [
+      { name = "Azure Zanculmarktum"; email = "zanculmarktum@gmail.com"; }
```
Comment (mkg20001) [inline pkgs/applications/networking/browsers/microsoft-edge/browser.nix:183]: this should go into maintainers/maintainer-list.nix in a seperate commit

**Review by mkg20001 — COMMENTED** [review]
It's failing to evaluate on ofborg, because aliases are being used

`at_spi2_atk` and the other at_spi2_ are aliases. Check pkgs/top-level/aliases.nix and replace the aliases with their actual values.

File: pkgs/applications/networking/browsers/microsoft-edge/browser.nix (hunk 1 above)
Comment (kuwii) [reply pkgs/applications/networking/browsers/microsoft-edge/browser.nix]: Updated in commit [c65edaa0285b7957654007333848c2ca877984f0](https://github.com/NixOS/nixpkgs/pull/161064/commits/c65edaa0285b7957654007333848c2ca877984f0).

**Comment by kuwii** [comment]
Updated to replace aliases with the real package names, and fixed some issues of desktop files.
@mkg20001 could you please take a look at it again? Thanks.

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -13953,4 +13953,16 @@
     github = "nigelgbanks";
     githubId = 487373;
   };
+  zanculmarktum = {
```
Comment (mkg20001) [inline maintainers/maintainer-list.nix:13956]: Each maintainer must be added in it's own seperate commit

The commit message must be as follow `maintainers: add <name>`

**Comment by mkg20001** [comment]
Can you squash the other commits besides the maintainer ones? When making a new package, you can just put everything into the init commit

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (kuwii) [reply maintainers/maintainer-list.nix:13956]: Updated commits. Sorry for the back and forth.

**Comment by kuwii** [comment]
Also ping @nixinator @aanderse @zanculmarktum

**Comment by hilaolu** [comment]
I have updated edge to 99.0.1150.46 via update.sh. But It doesn't work with wayland ozone.
```
microsoft-edge-stable  --enable-features=UseOzonePlatform --ozone-platform=wayland

[15885:15885:0318/225911.457140:ERROR:wayland_connection.cc(138)] Failed to load wayland client libraries.
[15885:15885:0318/225911.457171:ERROR:ozone_platform_wayland.cc(226)] Failed to initialize Wayland platform
[15885:15885:0318/225911.457175:ERROR:env.cc(225)] The platform failed to initialize.  Exiting.
```

**Comment by hello-smile6** [comment]
All checks have passed, are you going to merge?

**Review by hello-smile6 — APPROVED** [review]
Looks good!

**Comment by hello-smile6** [comment]
@mkg20001 You approving?

**Pull request merged**